### ISSUE TITLE
Fixed Issues with starting on index other than 0

### ISSUE THREAD

**Issue by paul-hart** [react-native-community/react-native-tab-view#516]
### Current behaviour

If you set the index to something other than 0, the TabBar control always shows the first tab as focused. I think it has to do with _handleOnLayout.

### Expected behaviour

Setting an index to 1 or higher should highlight that tab with the indicator below.

### About Fix

Note I used TapGestureHandler from react-native-gesture-handler but there are probably other ways. I just needed a "touch began" event to unlock the position. 

My understanding is the onLayout changes some of the Animation values and resets the position to 0 at some point. Rather than fixing that spot this is more of a hack / bandaid fix. 

### Code sample

```// @flow

import React from 'react';
import { Animated } from 'react-native';
import { TabViewAnimated } from 'react-native-tab-view';
import { TapGestureHandler } from 'react-native-gesture-handler';

// header

type Props = {
    ...TabViewAnimated.defaultProps
};

type State = {
    locked: Boolean
};

// class

export default class TabViewAnimatedEx extends TabViewAnimated<Props, State> {

    constructor(props) {
        super(props);
        // lock position at index
        this.state = {
            ...this.state,
            locked: true,
            position: new Animated.Value(props.navigationState.index)
        };
    }

    componentDidUpdate(oldProps) {
        const { navigationState } = this.props;
        const { index } = navigationState;
        const { panX } = this.state;
        const oldIndex = oldProps.navigationState.index;

        if (index !== oldIndex && panX._value !== 0) {
            this.resetResetCallback();
        }
    }

    resetResetCallback() {
        global.cancelAnimationFrame(this._resetCallback);
        this._resetCallback = global.requestAnimationFrame(() => this.resetOffset());
    }

    resetOffset() {
        const { navigationState } = this.props;
        const { layout } = this.state;

        const offset = navigationState.index * layout.width;
        this.state.offsetX.setValue(-offset);
        this.state.panX.setValue(0);
    }

    onTouchesBegan() {
        if (this.state.locked) {
            // unlock position
            const { panX, offsetX, layoutXY } = this.state;
            this.state.position = Animated.multiply(Animated.divide(Animated.add(panX, offsetX), layoutXY.x), -1);
            this.resetOffset();
            this.setState({ locked: false });
        }
    }

    render() {
        return <TapGestureHandler onBegan={() => this.onTouchesBegan()}>{super.render()}</TapGestureHandler>;
    }
}
```

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        | ios 11.3
| react-native          | 0.52.0
| react-native-tab-view | 0.0.74
| node                  | 9.4.0
| npm or yarn           | npm 5.8.0

**Comment by paul-hart** [comment]
Note this only seems to be an issue with iOS. I had to condition out the fix with android:

`// @flow

import React from 'react';
import { Animated, Platform } from 'react-native';
import { TabViewAnimated } from 'react-native-tab-view';
import { TapGestureHandler } from 'react-native-gesture-handler';

// header

type Props = {
    ...TabViewAnimated.defaultProps
};

type State = {
    isIOS: Boolean,
    locked: Boolean,
    position: Animated.Value
};

// class

export default class TabViewAnimatedEx extends TabViewAnimated<Props, State> {
    constructor(props) {
        super(props);

        const isIOS = Platform.OS === 'ios';

        // lock position at index
        this.state = {
            ...this.state,
            isIOS: isIOS,
            locked: isIOS,
            position: isIOS ? new Animated.Value(props.navigationState.index) : this.state.position
        };
    }

    componentDidUpdate(oldProps) {
        const { navigationState } = this.props;
        const { index } = navigationState;
        const { panX } = this.state;
        const oldIndex = oldProps.navigationState.index;

        if (index !== oldIndex && panX._value !== 0) {
            this.resetResetCallback();
        }
    }

    resetResetCallback() {
        global.cancelAnimationFrame(this._resetCallback);
        this._resetCallback = global.requestAnimationFrame(() => this.resetOffset());
    }

    resetOffset() {
        const { navigationState } = this.props;
        const { layout } = this.state;

        const offset = navigationState.index * layout.width;
        this.state.offsetX.setValue(-offset);
        this.state.panX.setValue(0);
    }

    onTouchesBegan() {
        if (this.state.locked) {
            // unlock position
            const { panX, offsetX, layoutXY } = this.state;
            this.state.position = Animated.multiply(Animated.divide(Animated.add(panX, offsetX), layoutXY.x), -1);
            this.resetOffset();
            this.setState({ locked: false });
        }
    }

    render() {
        const { isIOS } = this.state;

        if (isIOS) {
            return <TapGestureHandler onBegan={() => this.onTouchesBegan()}>{super.render()}</TapGestureHandler>;
        }

        return super.render();
    }
}
`

**Comment by botmane** [comment]
Hello,
Any update on this. I would like to start sometimes with index=1 but it doesn't seems to work.
Thanks

**Comment by Buddikazz** [comment]
Hi Is there any update on this for IOS

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
